CMakeLists.txt

```diff
@@ -219,6 +219,20 @@ if(CONFIG_LTO)
   add_link_options($<TARGET_PROPERTY:linker,lto_arguments>)
 endif()
 
+if(CONFIG_STD_C23)
```

Comment:
What would be a reason to not move this whole thing up here and get rid of the duplicate ifs?

https://github.com/zephyrproject-rtos/zephyr/blob/045a11fad19582b4434ea0dcf7b1cc01a4958767/CMakeLists.txt#L959-L987

Reply:
Done